source/codegen/service_helpers.py

```diff
@@ -284,6 +286,15 @@ def get_output_lookup_values(enum_data):
         out_value_format += f"{{{formated_value}, {i + 1}}},"
     return out_value_format
 
+def get_id_type_value(enum_data):
```

Comment:
All of these new method names are EXTREMELY generic. It's very difficult to tell from the names that this is codegen for a very specific feature that is only used to generate a few fields in codegen (i.e., it's not even used in the service cpp generator).

First we need a name for your feature and then we need to use it everywhere.

`type_mapped_enum`?
`enum_oneof_selector`?
`type_selector_enum`?

Also, some of these names are obviously insufficient. Like this one should mention that it's a map/lookup. Something like `generate_enum_oneof_selector_map`